apps/store/extensions/app/store-apis/apis/taxonomy.jag

```diff
@@ -35,123 +35,122 @@ require('/modules/store.js').exec(function (ctx) {
+                    jsonHeaderAccept = acceptHeader.indexOf('application/json');
+                    var options = req.getContent();
+
+                    try {
+                        if (jsonHeaderAccept >= 0 || acceptHeader == constants.ACCEPT_ALL) {
+                            var taxaPathList;
+                            var tenantDomain = options.tenant ? options.tenant : MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+                            tenantId = carbon.server.tenantId({domain: tenantDomain});
+                            options.tenantId = tenantId;
+                            options.username = username;
+
+                            if (options['terms']) {
+                                store.server.sandbox(options, function () {
+                                    taxaPathList = taxa.taxonomy.getNodesList(options['terms'],
+                                            defaultPagination(options, "startnode", 0),
+                                            defaultPagination(options, "endnode", 5));
+                                });
+                            } else {
+                                store.server.sandbox(options, function () {
+                                    taxaPathList = taxa.taxonomy.getNodesList("/*",
+                                            defaultPagination(options, "startnode", 0),
+                                            defaultPagination(options, "endnode", 5));
+                                });
+                            }
+
+
+                            response = responseProcessor.buildSuccessResponseForRxt(res, constants.STATUS_CODES.OK,
+                                    taxaPathList);
+                            return;
                         } else {
-                            store.server.sandbox(options, function () {
-                                taxaPathList = taxa.taxonomy.getNodesList("/*",
-                                        defaultPagination(options, "startnode", 0),
-                                        defaultPagination(options, "endnode", 5));
-                            });
+                            response = responseProcessor.buildErrorResponse(res, constants.STATUS_CODES.
+                                    UNSUPPORTED_MEDIATYPE, "error : Unsupported Media Type");
+                            return;
                         }
+                    } catch (e) {
 
-
-                        response = responseProcessor.buildSuccessResponseForRxt(res, constants.STATUS_CODES.OK,
-                                taxaPathList);
-                        return;
-                    } else {
-                        response = responseProcessor.buildErrorResponse(res, constants.STATUS_CODES.
-                                UNSUPPORTED_MEDIATYPE, "error : Unsupported Media Type");
+                        if (e.toString().indexOf("javax.xml.xpath.XPathExpressionException") > 0) {
```

Comment:
Thank you Sameera for noticing it,
This has been fixed with this PR https://github.com/wso2/carbon-store/pull/834